src/Apps/Artist/routes.tsx

```diff
@@ -87,42 +87,56 @@ export const routes: RouteConfig[] = [
     },
     query: graphql`
       query routes_ArtistTopLevelQuery($artistID: String!) @raw_response_type {
+        me {
+          id
+        }
         artist(id: $artistID) @principalField {
           ...ArtistApp_artist
           ...routes_Artist @relay(mask: false)
         }
       }
     `,
     render: ({ Component, props, match }) => {
-      if (Component && props) {
-        const { artist } = props as any
+      if (!(Component && props)) {
+        return null
+      }
 
-        if (!artist) {
-          return undefined
-        }
+      const { artist, me: user } = props as any
+      const { pathname } = match.location
 
-        const showArtistInsights =
-          showMarketInsights(artist) ||
-          (artist.insights && artist.insights.length > 0)
-        const hasArtistContent = hasOverviewContent(artist)
+      if (!artist) {
+        return undefined
+      }
 
-        const alreadyAtWorksForSalePath = match.location.pathname.includes(
-          `${artist.slug}/works-for-sale`
-        )
+      const showArtistInsights =
+        showMarketInsights(artist) ||
+        (artist.insights && artist.insights.length > 0)
+      const hasArtistContent = hasOverviewContent(artist)
 
-        const canShowOverview = showArtistInsights || hasArtistContent
+      const alreadyAtWorksForSalePath = pathname.includes(
+        `${artist.slug}/works-for-sale`
+      )
 
-        if (!canShowOverview && !alreadyAtWorksForSalePath) {
-          throw new RedirectException(`/artist/${artist.slug}/works-for-sale`)
-        }
+      const canShowOverview = showArtistInsights || hasArtistContent
 
-        return <Component {...props} />
+      if (user && !pathname.includes("/overview")) {
+        throw new RedirectException(`/artist/${artist.slug}/works-for-sale`)
       }
+
+      if (!user && pathname.includes("/overview") && canShowOverview) {
+        throw new RedirectException(`/artist/${artist.slug}`)
+      }
+
+      if (!canShowOverview && !alreadyAtWorksForSalePath) {
+        throw new RedirectException(`/artist/${artist.slug}/works-for-sale`)
+      }
+
+      return <Component {...props} />
     },
     children: [
       // Routes in tabs
       {
-        path: "/",
+        path: ":regexParam(\\overview)?",
```

Comment:
is `:regexParam` a special flag in found?

Reply:
It is, yeah. This is required to ensure that the overview route will correctly render for both `/` and `/overview` depending on the case.

Reply:
That's cool, i was wondering if Found supported this kind of thing